core/transformer/src/IModelTransformer.ts

```diff
@@ -1132,10 +1132,35 @@ export class IModelTransformer extends IModelExportHandler {
     return Semver.gt(`${schemaKey.version.read}.${schemaKey.version.write}.${schemaKey.version.minor}`, Schema.toSemverString(versionInTarget));
   }
 
+  private _longNamedSchemasMap = new Map<string, string>();
+
   /** Override of [IModelExportHandler.onExportSchema]($transformer) that serializes a schema to disk for [[processSchemas]] to import into
    * the target iModel when it is exported from the source iModel. */
   public override async onExportSchema(schema: ECSchemaMetaData.Schema): Promise<void> {
-    this.sourceDb.nativeDb.exportSchema(schema.name, this._schemaExportDir);
+    const ext = ".ecschema.xml";
```

Comment:
would it be simpler if you always serialized the schemas with the fixed names instead of only doing so if the names were too long.  

Serializing to disk is unnecessary over all with a custom context you could just wrap the call to get as xml string from the target iModel in a call to get the schema by name in the context.  Maybe this is not ideal because it would be harder to dump to disk for debug purposes.

Either way unless there is some real downside to using a single method for all schemas I think that is better.  The code just has fewer alternative paths to do the same thing

Reply:
I will look into removing the on disk paths entirely, and potentially redraft these PRs. That sounds much better.

Reply:
Unfortunately it's not currently possible without breaking the transformation service's  usage of the on-disk schemas. They modify some dynamic schemas while it's on disk and before importing. It is also not the bottleneck in transformation so isn't necessarily worthwhile.

I will however provide a way to access the path of the written file, in case it's name was truncated for being too long, for them to use because of this change where the written name is no longer guaranteed to be `` `${schema.name}.ecschema.xml` ``